hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCallLocalHandler.java

```diff
@@ -88,6 +93,10 @@ public void validate(@NonNull final QueryContext context) throws PreCheckExcepti
         final var maxGasLimit =
                 context.configuration().getConfigData(ContractsConfig.class).maxGasPerSec();
         validateTruePreCheck(requestedGas <= maxGasLimit, MAX_GAS_LIMIT_EXCEEDED);
+        final var intrinsicGas = gasCalculator.transactionIntrinsicGasCost(
+                org.apache.tuweni.bytes.Bytes.wrap(op.functionParameters().toByteArray()), false);
+        validateTruePreCheck(op.gas() >= intrinsicGas, INSUFFICIENT_GAS);
+
```

Comment:
Is this check duplicated in HevmStaticTransactionFactory:assertValidCall?

Reply:
It is but we'd like to address this before a transaction record gets created if possible